support-lib/jni/djinni_support.cpp

```diff
@@ -22,6 +22,7 @@
 #include <cstring>
 
 static_assert(sizeof(jlong) >= sizeof(void*), "must be able to fit a void* into a jlong");
+static_assert(sizeof(wchar_t) == 2 || sizeof(wchar_t) == 4, "wchar_t must be represented by UTF-16 or UTF-32 encoding");
```

Comment:
I'm curious whether in common usage a 4-byte wchar_t definitely indicates an assumption that UTF-32 is being used, as opposed to maybe UTF-16 which happens to have been copied into wider chars.  There's no enforced requirement either way, of course, any more than an 8-bit string is guaranteed to be UTF-8, rather than latin-1 or some other 8-bit charset.  I think the assumptions made here are as good as they need to be, I just wonder if they might bite someone later, and whether we'll really need a 3-way option to support 8/16/32 bit strings.